lib/ecto/migrator.ex

```diff
@@ -392,11 +392,12 @@ defmodule Ecto.Migrator do
 
   # This function will match directories passed into `Migrator.run`.
   defp migrations_for(migration_source) when is_binary(migration_source) do
-    query = Path.join(migration_source, "*")
+    query = Path.join([migration_source, "**", "*.exs"])
 
-    for entry <- Path.wildcard(query),
-        info = extract_migration_info(entry),
-        do: info
+    Path.wildcard(query)
+    |> Enum.map(&extract_migration_info/1)
+    |> Enum.filter(& &1)
+    |> Enum.sort_by(fn {version, _, _} -> version end)
```

Comment:
Just `|> Enum.sort()` is fine as we will end-up sorting by the first element of the tuple.